### PR TITLE
Add question about setTimeout

### PR DESCRIPTION
**Pull request by nsinghal12**
# Pull Request

Added a new JS question as to " At what time will the Settimeout() function be executed, if called at 4000 milliseconds?

### CODE REVIEW

File: questions/javascript-questions.md [hunk 1]
```diff
@@ -64,3 +64,4 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
 * What are the benefits of using `spread syntax` and how is it different from `rest syntax`?
 * How can you share code between files?
 * Why you might want to create static class members?
+* At what time will the Settimeout() function be executed, if called at 4000 milliseconds?
```
Comment (vvscode) [inline questions/javascript-questions.md]: maybe it worth to use the correct name of function? `setTimeOut` ? and (I'm not sure but) looks like in English `at 4000 milliseconds` sounds strange

File: questions/javascript-questions.md [hunk 2]
```diff
@@ -64,3 +64,4 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
 * What are the benefits of using `spread syntax` and how is it different from `rest syntax`?
 * How can you share code between files?
 * Why you might want to create static class members?
+* After how much time will the `callback` be executed if invoked as: `setTimeout(callback, 4000)`?
```
Comment (vvscode) [inline questions/javascript-questions.md]: looks like `after what time` should be

Comment (vvscode) [reply questions/javascript-questions.md]: and order of words

Comment (GMaiolo) [reply questions/javascript-questions.md]: Maybe `When will the callback be executed ...` ?

Comment (nsinghal12) [reply questions/javascript-questions.md]: It should be executed at time equivalent to 4000 milli seconds or more not at 3999 milliseconds.

Comment (roblarsen) [reply questions/javascript-questions.md]: I often ask a question about setTimout, in the context of a question about arrow functions. I first ask the interviewee to explain what setTimeout is and then I ask the following, which opens up a larger, potentially open-ended discussion. Trivia questions with a "right" answer are the kinds of things that people can cram for (by looking at a site like this.) Questions that serve as jumping off points are much less easy to prepare for, in advance. 

```suggestion
* When would you expect the `callback` function in the following setTimeout to be executed? 
`setTimeout(callback, 0)`?
```

**Pull request closed without merging**